src/analyses/constant_propagator.cpp

```diff
@@ -790,8 +790,10 @@ void constant_propagator_ait::replace(
     {
       if(it->get_other().get_statement() == ID_expression)
       {
+        auto c = to_code_expression(it->get_other());
         constant_propagator_domaint::partial_evaluate(
-          d.values, it->get_other(), ns);
+          d.values, c.expression(), ns);
+        it->set_other(c);
```

Comment:
Do this conditionally:
```
if(!constant_propagator_domaint::partial_evaluate(
    d.values, c.expression(), ns))
{
  it->set_other(c);
}
```
(or the shell-style `constant_propagator_domaint::partial_evaluate(d.values, c.expression(), ns) || it->set_other(c);`  👼)

Reply:
done